src/github.com/stellar/gateway/compliance/handlers/main.go

```diff
@@ -18,4 +21,21 @@ type RequestHandler struct {
 	SignatureSignerVerifier crypto.SignerVerifierInterface `inject:""`
 	StellarTomlResolver     stellartoml.ResolverInterface  `inject:""`
 	FederationResolver      federation.ResolverInterface   `inject:""`
+	NonceGenerator          NonceGeneratorInterface        `inject:""`
+}
+
+type NonceGeneratorInterface interface {
+	Generate() string
+}
+
+type NonceGenerator struct{}
+
+func (n *NonceGenerator) Generate() string {
+	return strconv.FormatInt(time.Now().UnixNano(), 10)
```

Comment:
This seems like something that we should ask @stanford-scs about.  Using time arbitrarily, without any commentary, strikes me as risky.  I imagine we should probably mix in some randomness as well, but I'm not the expert by any stretch :)